be/src/storage/lake/vacuum.cpp

```diff
@@ -1216,6 +1230,49 @@ static StatusOr<std::map<std::string, DirEntry>> find_orphan_data_files(FileSyst
         }
         LOG(INFO) << "Filtered with meta file: " << name << " (" << progress << '/' << meta_files.size() << ')';
     }
+    for (const auto& name : bundle_meta_files) {
+        auto location = join_path(metadata_root_location, name);
+        RandomAccessFileOptions opts{.skip_fill_local_cache = true};
+        ASSIGN_OR_RETURN(auto input_file, fs->new_random_access_file(opts, location));
+        ASSIGN_OR_RETURN(auto serialized_string, input_file->read_all());
+
+        auto file_size = serialized_string.size();
+        ASSIGN_OR_RETURN(auto bundle_metadata,
+                         TabletManager::parse_bundle_tablet_metadata(location, serialized_string));
+        for (const auto& tablet_page : bundle_metadata->tablet_meta_pages()) {
+            const PagePointerPB& page_pointer = tablet_page.second;
+            auto offset = page_pointer.offset();
+            auto size = page_pointer.size();
+            if (offset + size > file_size) {
+                LOG(WARNING) << "Invalid page pointer for tablet " << tablet_page.first << ", offset: " << offset
+                             << ", size: " << size << ", file size: " << file_size;
+                continue; // Skip this page if the pointer is invalid
```

Comment:
continue may cause segment files to be mistakenly deleted?

Reply:
You mean that we should skip the deletion when tablet meta is corrupt?

Reply:
done